perfkitbenchmarker/providers/openstack/os_virtual_machine.py

```diff
@@ -211,7 +211,7 @@ def _CheckNetworks(self):
 
   def _CheckFloatingIPNetworkExists(self, floating_network_name_or_id):
     network = self._CheckNetworkExists(floating_network_name_or_id)
-    if network['router:external'] != 'External':
+    if network['router:external'] != 'External' and not 'True':
```

Comment:
Do you mean _if network['router:external'] not in ('External', 'True')_?
because _not 'True'_ just evaluates to false

Reply:
The old version returns as 'External', and the new version returns as 'True', so if value is neither 'External' or 'True', then we need to let it fail.

Reply:
you will need to do _if network['router:external'] not in ('External', 'True')_ or  _if network['router:external'] != 'External' or network['router:external'] != 'True'_
What you wrote there currently means _if network['router:external'] != 'External' and if not 'True'_.
You can open an interactive python interpreter to fiddle and test. For example, set network['router:external'] = 'foobar' and see what result you get with each. :)
Since this can be confusing, you should also add a unit test.

Reply:
Thank you @jellyfishcake , it's my mistake. let me update patch.

Reply:
@jellyfishcake, Fix is updated and added unittest. It would be great if you can suggest better idea on creating benchmarkspec in unittest.

Reply:
Hi @jellyfishcake, could you please approve this change? This feature needs to be merged in order to have pkb openstack working on current API version.